generate/convert.go

```diff
@@ -354,7 +359,9 @@ func (g *generator) convertSelectionSet(
 		// GraphQL (and, effectively, JSON) requires that all fields with the
 		// same alias (JSON-name) must be the same (i.e. refer to the same
 		// field), so that's how we deduplicate.
-		if fieldNames[field.JSONName] {
+		// It's fine to have duplicate embeds (i.e. via named fragments), even
+		// ones with complicated overlaps, since they are separate types to us.
```

Comment:
It might be worth explicitly calling out that field.JSONName == "" indicates an embed.

Reply:
This will actually be further clarified in the next commit, so I'm gonna leave it to avoid conflicts.